src/utils/utils.ts

```diff
@@ -250,62 +233,77 @@ export const isNeedField = ({
     return false;
   }
 
-  const issueType = project.issuetypes.find(
-    ({ id }: JiraIssueType) => id === issueTypeId,
-  );
+  const issueType = project.issuetypes.find(({ id }) => id === issueTypeId);
 
   return issueType?.fields?.[fieldName] !== undefined;
 };
 
 export const jiraIssueToFormValues = (
-  issue: Record<string, any>,
-  usableFields: (Assignee | Attachment)[],
+  issue: IssueBean["fields"],
+  usableFields: FieldMeta[],
 ) => {
-  const values = Object.keys(issue).reduce(
-    (acc, key) => {
-      if (!usableFields.find((e) => e.key === key)) return acc;
+  const values = Object.keys(issue).reduce<JiraIssueSchema>((acc, key) => {
+    const usableField = usableFields.find((e) => e.key === key)
 
-      const usableField = usableFields.find((e) => e.key === key);
+    if (!usableField) {
+      return acc;
+    }
 
-      switch (usableField?.schema.type) {
-        case "user": {
-          acc[key] = {
-            id: issue[key as keyof IssueItem]?.accountId,
-          };
+    switch (usableField?.schema?.type) {
+      case "user": {
+        acc[key] = {
+          id: (issue[key] as UserBean)?.accountId,
+        };
 
-          break;
-        }
+        break;
+      }
 
-        case "date":
-        case "datetime": {
-          if (!issue[key as keyof IssueItem]) break;
-          acc[key] = new Date(issue[key as keyof IssueItem]);
+      case "date":
+      case "datetime": {
+        if (!issue[key]) break;
+        acc[key] = new Date(issue[key] as DateTime);
 
-          break;
-        }
+        break;
+      }
 
-        case "string": {
-          if (usableField.schema.system === "description") {
-            acc[key] = useAdfToPlainText(issue[key as keyof IssueItem]);
+      case "string": {
+        if (usableField.schema.system === "description" || usableField.schema.custom === FieldType.TEXT_PARAGRAPH) {
+          acc[key] = useAdfToPlainText(issue[key] as ADFEntity);
 
-            break;
-          }
+          break;
         }
-        // eslint-disable-next-line no-fallthrough
-        default:
-          acc[key] = issue[key as keyof IssueItem];
       }
+      // eslint-disable-next-line no-fallthrough
+      default:
+        acc[key] = issue[key];
+    }
 
-      return acc;
-    },
-    {} as Record<string, unknown>,
-  );
+    return acc;
+  }, {} as JiraIssueSchema);
 
   return values;
 };
 
-export const objectToStringWithoutBraces = (obj: Record<string, string>) => {
+export const errorToStringWithoutBraces = (
+  obj: Record<string, string>,
+  metaMap: Record<FieldMeta["key"], FieldMeta>,
+) => {
   return Object.entries(obj)
-    .map(([key, value]) => `${key}: ${value}`)
-    .join(" ");
+    .map(([key, value]) => `${metaMap[key]?.name || key}: ${value}`)
+    .join("; ");
+};
+
+export const getLayout = (mapping: Settings["mapping"]): Layout => {
+  const settingsLayout = mapping || "{}";
+  let data: Layout;
+
+  try {
+    data = JSON.parse(settingsLayout) as Layout;
```

Comment:
Should validate before returning.

Reply:
added checking `if (Array.isArray(data.detailView) && Array.isArray(data.listView)) ...`